geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreFactory.scala

```diff
@@ -24,15 +27,31 @@ class HBaseDataStoreFactory extends DataStoreFactorySpi {
 
   import HBaseDataStoreFactory.Params._
 
+  // TODO: can we have multiple connections in a single JVM?
+  private class Key(val connection: Connection, val config: HBaseDataStoreConfig) {
+    override def hashCode(): Int = config.hashCode()
+    override def equals(obj: scala.Any): Boolean = config.equals(obj)
+  }
+
+  private val dsCache =
+    CacheBuilder.newBuilder().build(
+      new CacheLoader[Key, HBaseDataStore] {
+        override def load(k: Key): HBaseDataStore = {
+          new HBaseDataStore(k.connection, k.config)
```

Comment:
Just caching the connection for now.  Will have to think through caching relations for a subsequent ticket.